src/open_samus_returns_rando/specific_patches/door_patches.py

```diff
@@ -569,17 +617,18 @@ def add_custom_shields(editor: PatcherEditor, new_shield: NewShield) -> None:
         custom_shield.raw["components"]["LIFE"]["fields"]["fTimeToStartDisolve"]["value"] = 0.2
 
 
-def patch_doors(editor: PatcherEditor, door_patches: list[dict]) -> None:
+def patch_doors(editor: PatcherEditor, door_patches: list[dict], custom_doors: list[dict]) -> None:
+    # Patch static door changes
     _static_door_patches(editor)
-    for new_shield in new_shields:
-        add_custom_shields(editor, new_shield)
-    door_patcher = DoorPatcher(editor)
 
-    # small hack to eliminate duplicates (randovania exports everything duplicated)
-    import json
+    # Add custom door actors
+    add_custom_doors(editor, custom_doors)
 
-    set_of_jsons = {json.dumps(d, sort_keys=True) for d in door_patches}
-    door_patches_set = [json.loads(t) for t in set_of_jsons]
+    # Create new door types
+    for new_shield in new_shields:
+        add_custom_shields(editor, new_shield)
 
-    for door in door_patches_set:
```

Comment:
I prefer to keep it:
1. It's much faster to not patch every door twice
2. I'm pretty sure the way I wrote the minimap stuff will not work with the duplicates. Do you tested this?

Reply:
My current wip branch on the rdv side which adds all the new doors makes changes to the door_patches function to not export duplicates. So the functionality is the same as before. The pr will be live soon so you'll seen it then.